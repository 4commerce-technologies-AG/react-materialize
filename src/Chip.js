import React from 'react';
import PropTypes from 'prop-types';

const Chip = ({ children, close, ...props }) => {
  return (
    <div className="chip" {...props}>
      {children}
      {close ? <i className="close material-icons">close</i> : null}
    </div>
  );
};

Chip.propTypes = {
  children: PropTypes.node,
  /**
   * Shows a close icon
   */
  close: PropTypes.bool
};

Chip.defaultProps = {
  close: false
};

export default Chip;
