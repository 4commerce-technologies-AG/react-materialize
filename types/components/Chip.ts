import * as React from 'react';
import { SharedBasic, AnyFn } from './utils';
import { AutocompleteOptions } from './Autocomplete';

type ChipItem = {[key: string]: string};

export interface ChipOptions {
  data?: ChipItem[];
  placeholder?: string;
  secondaryPlaceholder?: string;
  autocompleteOptions?: AutocompleteOptions;
  limit?: number;
  onChipAdd?: AnyFn;
  onChipSelect?: (item: React.ReactNode) => void;
  onChipDelete?: AnyFn;
};

export interface ChipProps extends SharedBasic {
  close?: boolean;
  closeIcon?: React.ReactNode;
  options?: ChipOptions;
};

/**
 * React Materialize: Chip
 */
declare const Chip: React.FC<ChipProps>

export default Chip;
