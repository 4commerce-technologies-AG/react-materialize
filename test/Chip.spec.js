import React from 'react';
import { shallow } from 'enzyme';
import { assert } from 'chai';
import Chip from '../src/Chip';

describe('<Chip />', () => {
  let wrapper;

  test('should render', () => {
    wrapper = shallow(<Chip />);
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts a close prop', () => {
    wrapper = shallow(<Chip close />);
    assert(wrapper.find('i.material-icons.close').length, 'with a close icon');
  });

  test('accepts children', () => {
    let child = <p className="child">Child</p>;
    wrapper = shallow(<Chip close>{child}</Chip>);
    assert(wrapper.contains(child), 'should not be empty');
  });
});
