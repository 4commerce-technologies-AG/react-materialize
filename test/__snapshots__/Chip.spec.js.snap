// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip /> should render 1`] = `
<div
  className="chip"
/>
`;
